app/controllers/health_check_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class HealthCheckController < ApplicationController
+  def all
+    response = 'success'
+    @cache_expire = 10.seconds
+
+    begin
+      cache_check
+      database_check
+    rescue StandardError => e
+      response = "Health Check Failure: #{e}"
+    end
+
+    render(plain: response)
+  end
+
+  private
+
+  def cache_check
+    raise 'Unable to write to cache' unless Rails.cache.write('__health_check_cache_write__', 'true', expires_in: @cache_expire)
+
+    raise 'Unable to read from cache' unless Rails.cache.read('__health_check_cache_write__') == 'true'
```

Comment:
Why this one does not have expiration like the prevous one?

And do we need to compare to String 'true'

Reply:
The expiry is only used for a cache write, not a cache read. 

The cache read just checks the the value that was stored in the cache from the previous step. It can be set to any string as long as the string is the same in the write and the read

Reply:
True, I misread the intention of the test

Do we need to verify read?

Reply:
Is this supposed to check the connection to redis? Note that the redis connection does not use the Rails cache, and the rails cache is just using the default in-process memory mode, so this doesn't really tell anything.

Reply:
If you want to test the redis connection you should do something like
```ruby
RedisStore.with_connection do |redis|
  # run some redis commands here, the 'redis' var is a connection
end
```